app/utils.py

```diff
@@ -1176,8 +1177,19 @@ class XsrfTool(object):
     # XSRF tokens expire after 4 hours.
     TOKEN_EXPIRATION_TIME = 60 * 60 * 4
 
+    # Characters with which to choose a key if it's not set yet.
+    TOKEN_CHARACTER_SET = string.letters + string.digits
+
     def __init__(self):
-        self._key = config.get('xsrf_token_key').encode('utf-8')
+        configured_key = config.get('xsrf_token_key')
+        if configured_key:
+            # config.get returns unicode, but hmac is going to want a str
+            self._key = configured_key.encode('utf-8')
+        else:
+            configured_key = ''.join([
+                XsrfTool.TOKEN_CHARACTER_SET for _ in range(20)])
```

Comment:
Maybe you forgot to call random choice function here? Looks like this always generates the same key, 20 time repeat of 'abc...789'. Also, consider using utils.generate_random_key instead:
https://github.com/google/personfinder/blob/17f793f3fee56eb6b5cdfa6027d5262d2d8e12de/app/utils.py#L615

Reply:
Whoa, yeah, you're right, I dropped random.choice by accident when I set up the character set constant. Thanks for catching that! I changed it to call generate_random_key (PR #559).